### ISSUE TITLE
Brightness + Color effect doesn't work the same as Scratch 2

### ISSUE THREAD

**Issue by paulkaplan** [LLK/scratch-render#142]
### Expected Behavior

_Please describe what should happen_
Scratch 2, if you set brightness effect to 100 and color effect to 50, this is what you get
![image](https://cloud.githubusercontent.com/assets/654102/26692589/3c5400fc-46cf-11e7-80f4-676881de25e2.png)

### Actual Behavior

_Describe what actually happens_

In Scratch 3, you get all white (I added a background so that you can see it). 
![image](https://cloud.githubusercontent.com/assets/654102/26692619/61a106e8-46cf-11e7-8480-584f2b74aed1.png)

---

This bug is a simplification of a problem I had when opening this project in scratch 3 https://scratch.mit.edu/projects/1270935/
It should look like this
![image](https://cloud.githubusercontent.com/assets/654102/26692693/9c0eed5e-46cf-11e7-80e9-d713cbbe6091.png)

And instead looked like this 
![image](https://cloud.githubusercontent.com/assets/654102/26692699/a0903a36-46cf-11e7-8263-966a0203dd1d.png)

---
cc @cwillisf

**Comment by BryceLTaylor** [comment]
I'm still seeing some strangeness with Brightness but it's a little different from what's listed above.  It looks like project 1270935 is working properly.  Also, brightnesses between -100 and 0 are behaving as expected.

For brightnesses between 0 and 100 it looks like it is getting brighter, but not all the way white as it does in 2.0.

Here it is in 2.0:
![brightnesstest2 0](https://user-images.githubusercontent.com/1088291/50648609-b24e0e80-0f49-11e9-94c3-77477cff1da7.gif)

And here it is in 3.0:
![brightnesstest3 0](https://user-images.githubusercontent.com/1088291/50648621-b843ef80-0f49-11e9-938e-c22107cd0ced.gif)

**Comment by fsih** [comment]
I found something weird. If you use just the brightness effect on a sprite in scratch 2, it looks like this (0 to 100)
<img width="474" alt="screen shot 2019-01-09 at 22 32 37" src="https://user-images.githubusercontent.com/2855464/50944490-9cb37a00-145e-11e9-872e-4c01d42bef1f.png">

However, if you also use the color effect on any sprite, all of the sprites change to look like this:
<img width="475" alt="screen shot 2019-01-09 at 22 32 14" src="https://user-images.githubusercontent.com/2855464/50944507-b359d100-145e-11e9-8ae7-c4690a496aa8.png">

And that's what the brightness effect looks like in Scratch 3.0 right now
<img width="470" alt="screen shot 2019-01-09 at 22 29 56" src="https://user-images.githubusercontent.com/2855464/50944515-c2408380-145e-11e9-8425-32d776fa6a5c.png">

@cwillisf when the color effect broke things last time you said it was something to do with the rendering engine?

**Comment by fsih** [comment]
Found a relevant thread with useful code links: https://github.com/LLK/scratch-flash/issues/570

Note though that HSV vs HSL isn't the problem here though. Here is 1) the current brightness effect, 2) the brightness effect using HSL, 3) the brightness effect in Scratch 2. Neither 1 nor 2 match 3.
<img width="1301" alt="screen shot 2019-01-09 at 22 30 45" src="https://user-images.githubusercontent.com/2855464/50948037-d5f3e600-146e-11e9-9266-c365c4785b71.png">

**Comment by towerofnix** [comment]
Maybe relevant: https://scratch.mit.edu/projects/269088736/#comments-105915881

TL;DR a positive brightness effect doesn't seem to have an impact on a costume whose fill colors are already 100% brightness (i.e. the sliders show brightness = 100). I think the brightness effect in 2.0 is more like "mix with white (or black)" rather than "adjust brightness component"? I haven't done testing though.

**Comment by fsih** [comment]
Projects mostly use the brightness effect to transition to and from white and black, so I think it would make the most sense to match the 2D renderer of Scratch 2. Sprites becoming white was also the original behavior before we had to add the 3D renderer, and the bug LLK/scratch-flash#570 was created because the 3D renderer was "wrong": https://scratch.mit.edu/discuss/topic/73118/

**Comment by thisandagain** [comment]
@cwillisf @fsih Matching the behavior of the 2D renderer makes sense to me. Thanks for the investigation!
